DAOS-16072 test: Support non-root user client commands

phender opened the pull request:
Support running commands on client hosts w/o sudo permissions.
- Add new launch.py command line arguments to specify a non-root daos_agent user
- Disable using sudo for fault injection file removal

Skip-unit-tests: true
Skip-fault-injection-test: true
Skip-func-hw-test: true
Test-tag: test_valid_labels test_launch_failures

Required-githooks: true

### Before requesting gatekeeper:

* [ ] Two review approvals and any prior change requests have been resolved.
* [ ] Testing is complete and all tests passed or there is a reason documented in the PR why it should be force landed and forced-landing tag is set.
* [ ] `Features:` (or `Test-tag*`) commit pragma was used or there is a reason documented that there are no appropriate tags for this PR.
* [ ] Commit messages follows the guidelines outlined [here](https://daosio.atlassian.net/wiki/spaces/DC/pages/11133911069/Commit+Comments).
* [ ] Any tests skipped by the ticket being addressed have been run and passed in the PR.

### Gatekeeper:

* [ ] You are the appropriate gatekeeper to be landing the patch.
* [ ] The PR has 2 reviews by people familiar with the code, including appropriate owners.
* [ ] Githooks were used. If not, request that user install them and check copyright dates.
* [ ] Checkpatch issues are resolved.  Pay particular attention to ones that will show up on future PRs.
* [ ] All builds have passed.  Check non-required builds for any new compiler warnings.
* [ ] Sufficient testing is done. Check feature pragmas and test tags and that tests skipped for the ticket are run and now pass with the changes.
* [ ] If applicable, the PR has addressed any potential version compatibility issues.
* [ ] Check the target branch.   If it is master branch, should the PR go to a feature branch?  If it is a release branch, does it have merge approval in the JIRA ticket.
* [ ] Extra checks if forced landing is requested
  * [ ] Review comments are sufficiently resolved, particularly by prior reviewers that requested changes.
  * [ ] No new NLT or valgrind warnings.  Check the classic view.
  * [ ] Quick-build or Quick-functional is not used.
* [ ] Fix the commit message upon landing. Check the standard [here](https://daosio.atlassian.net/wiki/spaces/DC/pages/11133911069/Commit+Comments). Edit it to create a single commit. If necessary, ask submitter for a new summary.

phender committed:
DAOS-16072 test: Support non-root user client commands

Support running commands on client hosts w/o sudo permissions.
- Add new launch.py command line arguments to specify a non-root
  daos_agent user

Skip-unit-tests: true
Skip-fault-injection-test: true
Skip-func-hw-test: true
Test-tag: test_valid_labels test_launch_failures

Required-githooks: true

Signed-off-by: Phil Henderson <phillip.henderson@intel.com>

daosbuild1 commented:
Test stage Functional on EL 8.8 completed with status FAILURE.  https://build.hpdd.intel.com//job/daos-stack/job/daos/view/change-requests/job/PR-14670/1/execution/node/798/log

phender committed:
Additional fixes:

- Disable using sudo for fault injection file removal

Skip-unit-tests: true
Skip-fault-injection-test: true
Skip-func-hw-test: true
Test-tag: test_valid_labels test_snapshots test_container_label_valid

Required-githooks: true

Signed-off-by: Phil Henderson <phillip.henderson@intel.com>

phender committed:
Revert unitended change to get_clush_command.

Skip-unit-tests: true
Skip-fault-injection-test: true
Skip-func-hw-test: true
Test-tag: test_valid_labels test_snapshots test_container_label_valid

Required-githooks: true

Signed-off-by: Phil Henderson <phillip.henderson@intel.com>

phender committed:
Adding passing user to all ExecutableCommand classes.

Skip-unit-tests: true
Skip-fault-injection-test: true
Skip-func-hw-test: true
Test-tag: test_valid_labels test_snapshots test_container_label_valid

Required-githooks: true

Signed-off-by: Phil Henderson <phillip.henderson@intel.com>

daosbuild1 commented:
Test stage Functional on EL 8.8 completed with status FAILURE.  https://build.hpdd.intel.com//job/daos-stack/job/daos/view/change-requests/job/PR-14670/4/execution/node/798/log

phender committed:
Fix setting DAOS_AGENT_DRPC_DIR.

Skip-unit-tests: true
Skip-fault-injection-test: true
Skip-func-hw-test: true
Test-tag: test_valid_labels test_snapshots test_container_label_valid

Required-githooks: true

Signed-off-by: Phil Henderson <phillip.henderson@intel.com>

phender committed:
Renamed run_utils.py run_command to issue_command.

Added passing agent_user into DaosAgentCommand.

Skip-unit-tests: true
Skip-fault-injection-test: true
Skip-func-hw-test: true
Test-tag: test_valid_labels test_snapshots test_container_label_valid

Required-githooks: true

Signed-off-by: Phil Henderson <phillip.henderson@intel.com>

phender committed:
Fix for test-specific temp dir.

Skip-unit-tests: true
Skip-fault-injection-test: true
Skip-func-hw-test: true
Test-tag: test_valid_labels test_snapshots test_container_label_valid

Required-githooks: true

Signed-off-by: Phil Henderson <phillip.henderson@intel.com>

daosbuild1 commented:
Test stage Functional on EL 8.8 completed with status FAILURE.  https://build.hpdd.intel.com//job/daos-stack/job/daos/view/change-requests/job/PR-14670/7/execution/node/799/log

phender committed:
Adding harness/coverage.py test.

Skip-unit-tests: true
Skip-fault-injection-test: true
Skip-func-hw-test: true
Test-tag: test_valid_labels test_snapshots test_container_label_valid test_basic_coverage

Required-githooks: true

Signed-off-by: Phil Henderson <phillip.henderson@intel.com>

daltonbohning commented on src/tests/ftest/harness/coverage.py:
```diff
@@ -0,0 +1,40 @@
+    def test_basic_coverage(self):
+        """Test basic harness coverage by starting servers, agents, dfuse, and running ior.
+
+        :avocado: tags=all,daily_regression
```
I think better to put as manual or in weekly

daltonbohning commented on src/tests/ftest/harness/coverage.yaml:
```diff
@@ -0,0 +1,38 @@
+ior: &ior_base
+  ppn: 4
+  api: DFS
+  transfer_size: 64K
+  block_size: 128K
```
This could potentially be super slow due to the VM slowness issue. I'd recommend reducing drastically to like 1K or less, but even that could be slow

daosbuild1 commented:
Test stage Functional on EL 8.8 completed with status FAILURE.  https://build.hpdd.intel.com//job/daos-stack/job/daos/view/change-requests/job/PR-14670/8/execution/node/798/log

phender committed:
Fix new test.

Skip-unit-tests: true
Skip-fault-injection-test: true
Skip-func-hw-test: true
Test-tag: test_valid_labels test_snapshots test_container_label_valid test_basic_coverage

Required-githooks: true

Signed-off-by: Phil Henderson <phillip.henderson@intel.com>

daosbuild1 commented:
Test stage Functional on EL 8.8 completed with status FAILURE.  https://build.hpdd.intel.com//job/daos-stack/job/daos/view/change-requests/job/PR-14670/9/execution/node/814/log

phender committed:
Remove additional sudo commands.

Skip-unit-tests: true
Skip-fault-injection-test: true
Skip-func-hw-test: true
Test-tag: test_valid_labels test_snapshots test_container_label_valid test_basic_coverage

Required-githooks: true

Signed-off-by: Phil Henderson <phillip.henderson@intel.com>

phender committed:
Additoinal change:

- Adding support for specifying a specific LD_LIBRARY_PATH to use in the
daos_server config environment variables for eeach engine.

Skip-unit-tests: true
Skip-fault-injection-test: true
Skip-func-hw-test: true
Test-tag: test_valid_labels test_snapshots test_container_label_valid test_basic_coverage

Required-githooks: true

Signed-off-by: Phil Henderson <phillip.henderson@intel.com>

phender committed:
Updates to daos_server LD_LIBRARY_PATH.

Skip-unit-tests: true
Skip-fault-injection-test: true
Skip-func-hw-test: true
Test-tag: test_valid_labels test_snapshots test_container_label_valid test_basic_coverage

Required-githooks: true

Signed-off-by: Phil Henderson <phillip.henderson@intel.com>

phender committed:
Cleanup run_utils.py

- Initial removal of run_command

Skip-unit-tests: true
Skip-fault-injection-test: true
Skip-func-hw-test: true
Test-tag: pr test_valid_labels test_snapshots test_container_label_valid test_basic_coverage HarnessUnitTest

Required-githooks: true

Signed-off-by: Phil Henderson <phillip.henderson@intel.com>

daosbuild1 commented:
Test stage Build on EL 8 completed with status FAILURE.  https://build.hpdd.intel.com//job/daos-stack/job/daos/view/change-requests/job/PR-14670/13/execution/node/284/log

daosbuild1 commented:
Test stage Build RPM on EL 8 completed with status FAILURE.  https://build.hpdd.intel.com//job/daos-stack/job/daos/view/change-requests/job/PR-14670/13/execution/node/398/log

daosbuild1 commented:
Test stage Build RPM on EL 9 completed with status FAILURE.  https://build.hpdd.intel.com//job/daos-stack/job/daos/view/change-requests/job/PR-14670/13/execution/node/395/log

daosbuild1 commented:
Test stage Build RPM on Leap 15.5 completed with status FAILURE.  https://build.hpdd.intel.com//job/daos-stack/job/daos/view/change-requests/job/PR-14670/13/execution/node/327/log

daosbuild1 commented:
Test stage Build on Leap 15.5 with Intel-C and TARGET_PREFIX completed with status FAILURE.  https://build.hpdd.intel.com//job/daos-stack/job/daos/view/change-requests/job/PR-14670/13/execution/node/494/log

daosbuild1 commented:
Test stage Build DEB on Ubuntu 20.04 completed with status FAILURE.  https://build.hpdd.intel.com//job/daos-stack/job/daos/view/change-requests/job/PR-14670/13/execution/node/383/log

phender committed:
Sync changes.

Skip-unit-tests: true
Skip-fault-injection-test: true
Skip-func-hw-test: true
Test-tag: pr test_valid_labels test_snapshots test_container_label_valid test_basic_coverage HarnessUnitTest DdbTest test_container_basics

Required-githooks: true

Signed-off-by: Phil Henderson <phillip.henderson@intel.com>

phender committed:
Merge branch 'master' into pahender/DAOS-16072_2

Skip-unit-tests: true
Skip-fault-injection-test: true
Skip-func-hw-test: true
Test-tag: pr test_valid_labels test_snapshots test_container_label_valid test_basic_coverage HarnessUnitTest DdbTest test_container_basics

Required-githooks: true

Signed-off-by: Phil Henderson <phillip.henderson@intel.com>

phender committed:
Merge branch 'master' into pahender/DAOS-16072_2

Skip-unit-tests: true
Skip-fault-injection-test: true
Skip-func-hw-test: true
Test-tag: pr test_valid_labels test_snapshots test_container_label_valid test_basic_coverage HarnessUnitTest DdbTest test_container_basics

Required-githooks: true

Signed-off-by: Phil Henderson <phillip.henderson@intel.com>

daosbuild1 commented:
Test stage Functional on EL 8.8 completed with status FAILURE.  https://build.hpdd.intel.com//job/daos-stack/job/daos/view/change-requests/job/PR-14670/16/execution/node/823/log
